### PR TITLE
fix(earn): use statsig dynamic config for app fee

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Uses the Statsig dynamic config to determine if swap fees should be waived.

### Test plan

- [x] Unit tests updated
- [x] Tested locally with current backend

### Related issues

Part of ACT-1379

### Backwards compatibility

Yes

### Network scalability

N/A

### CODE REVIEW

File: src/positions/saga.ts [hunk 1]
```diff
@@ -258,14 +259,18 @@ export function* handleEnableHooksPreviewDeepLink(
...
     getHooksApiFunctionUrl(hooksApiUrl, 'triggerShortcut').toString(),
     {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json',
       },
-      body: JSON.stringify(bodyJson),
+      body: JSON.stringify({
+        ...bodyJson,
+        enableSwapFee: enableAppFee,
```
Comment (satish-ravi) [inline src/positions/saga.ts:272]: This should be specific to swap-deposit. Can we do it earn/prepareTransactions?

Comment (MuckT) [reply src/positions/saga.ts]: Fixed in bf7459b287507559c1eed277cf6b1dd7046798ef!

File: src/earn/prepareTransactions.ts [hunk 2]
```diff
@@ -64,6 +70,7 @@ export async function prepareDepositTransactions({
       shortcutId,
       ...args,
       ...pool.shortcutTriggerArgs?.[shortcutId],
+      enableSwapFee,
```
Comment (satish-ravi) [inline src/earn/prepareTransactions.ts:73]: I guess the triggerShortcut call will discard this for deposit, but to be clean, you can set this conditionally in args for swap-deposit, right next to swapFromToken

Comment (MuckT) [reply src/earn/prepareTransactions.ts]: Fixed in 5140644529608dc9fe9a86f5d4ecca862232562c!

**Review by satish-ravi — APPROVED** [review]

**Pull request merged**